cmd/example-posix/main.go

```diff
@@ -47,19 +53,50 @@ func main() {
 
 	// Read log public key from file or environment variable
 	var pubKey string
+	var err error
 	if len(*pubKeyFile) > 0 {
-		k, err := os.ReadFile(*pubKeyFile)
+		pubKey, err = getKeyFile(*pubKeyFile)
 		if err != nil {
-			klog.Exitf("failed to read public_key file: %q", err)
+			klog.Exitf("Unable to get public key: %q", err)
 		}
-		pubKey = string(k)
 	} else {
-		pubKey = os.Getenv("SERVERLESS_LOG_PUBLIC_KEY")
+		pubKey = os.Getenv("LOG_PUBLIC_KEY")
 		if len(pubKey) == 0 {
-			klog.Exit("supply public key file path using --public_key or set SERVERLESS_LOG_PUBLIC_KEY environment variable")
+			klog.Exit("Supply public key file path using --public_key or set LOG_PUBLIC_KEY environment variable")
+		}
+	}
+	// Read log private key from file or environment variable
+	var privKey string
+	if len(*privKeyFile) > 0 {
+		privKey, err = getKeyFile(*privKeyFile)
+		if err != nil {
+			klog.Exitf("Unable to get private key: %q", err)
+		}
+	} else {
+		privKey = os.Getenv("LOG_PRIVATE_KEY")
+		if len(privKey) == 0 {
+			klog.Exit("Supply private key file path using --private_key or set LOG_PRIVATE_KEY environment variable")
 		}
 	}
 
+	var cpNote note.Note
+	s, err := note.NewSigner(privKey)
+	if err != nil {
+		klog.Exitf("Failed to instantiate signer: %q", err)
+	}
+	if *initialise {
+		if err := os.MkdirAll(*storageDir, dirPerm); err != nil {
+			klog.Exitf("Failed to create log directory: %q", err)
+		}
+		cp := fmtlog.Checkpoint{
+			Hash: rfc6962.DefaultHasher.EmptyRoot(),
+		}
+		if err := signAndWrite(&cp, cpNote, s, *storageDir); err != nil {
```

Comment:
Probably worth trying to remove `signAndWrite` and use `writeCP` (or similar) if possible - currently initialising the log will use `*origin` for the `CP.Origin` but updates to the log will use `s.Name()`.